src/qtt/instrument_drivers/gates.py

```diff
@@ -246,9 +246,22 @@ def get_boundaries(self):
             boundaries[gate] = param.vals.valid_values
         return boundaries
 
+    def restrict_boundaries(self, gate_boundaries):
+        """ Restrict boundaries values that can be set on the gates.
+
+        Args:
+            gate_boundaries (dict): a range of allowed values per parameter.
+        """
+        new_boundaries = self.get_boundaries()
+
+        for g, bnds in gate_boundaries.items():
```

Comment:
g should be renamed to gate, bnds to boundary. It's all in the name, keep it as clear as possible.